service/history/execution/context.go

```diff
@@ -45,10 +45,29 @@ const (
 	defaultRemoteCallTimeout = 30 * time.Second
 )
 
-var (
-	// ErrConflict is exported temporarily for integration test
-	ErrConflict = errors.New("conditional update failed")
-)
+type conflictError struct {
+	Cause error
+}
+
+func (e *conflictError) Error() string {
+	return fmt.Sprintf("conditional update failed: %s", e.Cause.Error())
+}
+
+func (e *conflictError) Unwrap() error {
+	return e.Cause
+}
+
+// NewConflictError is only public because it used in workflow/util_test.go
+// TODO: refactor those tests
+func NewConflictError(cause error) error {
+	return &conflictError{cause}
+}
```

Comment:
personal preference: require a `*testing.T` argument for any test-only code.  even if it's unused, it makes it rather unambiguous that it's not intended for any other kind of use.

Reply:
That is actually a neat idea 👍 . I'm going to do that from now on.